core/string/ustring.cpp

```diff
@@ -2575,6 +2575,72 @@ int64_t String::to_int(const wchar_t *p_str, int p_len) {
 	return integer * sign;
 }
 
+uint64_t String::to_uint64(bool p_clamp) const {
+	int str_length = length();
+	if (str_length == 0) {
+		return 0;
+	}
+
+	int start = 0;
+	// Skip leading whitespace
+	while (start < str_length &&
+			(is_whitespace(operator[](start)) || operator[](start) == '+')) {
+		start++;
+	}
+
+	// Check for negative sign
+	if (start < str_length && operator[](start) == '-') {
+		if (!p_clamp) {
+			ERR_FAIL_V_MSG(0,
+					"Cannot represent " + *this + " as a 64-bit unsigned integer, since the value is negative.");
+		}
+		return 0;
+	}
+
+	uint64_t integer = 0;
+	for (int i = start; i < str_length; i++) {
+		char32_t c = operator[](i);
+		if (!is_digit(c)) {
+			break;
+		}
+
+		bool overflow = (integer > UINT64_MAX / 10) ||
+				(integer == UINT64_MAX / 10 && (c > '5')); // 18446744073709551615 is UINT64_MAX
+		if (overflow) {
+			if (!p_clamp) {
+				ERR_FAIL_V_MSG(UINT64_MAX,
+						"Cannot represent " + *this + " as a 64-bit unsigned integer, since the value is too large.");
+			}
+			return UINT64_MAX;
+		}
+		integer *= 10;
+		integer += c - '0';
+	}
+	return integer;
+}
+
+uint64_t String::to_uint64(const char *p_str, int p_len) {
```

Comment:
These functions are unnecessarily inefficient, but they're designed after the current `to_int` function, so it's fine. Tools to make them efficient are not fully in place yet so I'll think this is appropriate for now.

Reply:
Honestly I wanted to mostly just keep in line with what was there mostly to minimise discussion.

Reply:
Yes, that was the correct decision. Feel free to close this one.